run_code_block_tests.sh

```diff
@@ -1,4 +1,4 @@
 set -euo pipefail
 
-find extracted -name "*.shell_session" -exec echo "running {}" \; -exec ./convert_shell_session_to_pexpect_test.py {} \;
-find extracted -name "test_*"  -exec echo "running {}" \; -execdir {} \;
+find extracted -name "*.shell_session" | xargs -I{} sh -c 'echo "running {}"; ./convert_shell_session_to_pexpect_test.py {};'
+find extracted -name "test_*" | xargs -I{} sh -c 'echo "running {}"; cd `dirname {}`; ./`basename {}`'
```

Comment:
Hey @cheriimoya, is there a better way to do this?

Reply:
this looks reasonable to me. the only thing you could change is to use finds `-print0` and xargs `-0` expression like this: `find -name "blah" -print0 | xargs -0 echo do stuff`

for an explanation, check out the second point on this list https://www.theunixschool.com/2012/11/examples-find-command-xargs-linux.html , pretty much to allow spaces in filenames etc